DelegateUIKit/ViewController.swift

```diff
@@ -7,13 +7,77 @@
 
 import UIKit
 
-class ViewController: UIViewController {
+//the viewController has to be conform to the Delegete we want to listen to
+class ViewController: UIViewController, ProductSelectionDelegate {
+    
+    let productImageView = UIImageView()
+    let productNameLabel = UILabel()
+    let chooseProductButton = UIButton()
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        setupUI()
     }
 
+    //this method knows what to do and is listening to the event (button click)
+    func didSelectProduct(name: String, imageName: String) {
```

Comment:
It's nothing wrong to add this delegate here but its usually best practice to add the delegate in an extension of the view controller.

Like this

```
extension ViewController: ProductSelectionDelegate {
    func didSelectProduct(name: String, imageName: String) {
    // do stuff here
    }
}
```